### ISSUE TITLE
hashiwrapper strips useful python stacktrace info on hard failures

### ISSUE THREAD

**Issue by drewmullen** [TerryHowe/ansible-modules-hashivault#182]
https://github.com/TerryHowe/ansible-modules-hashivault/blob/bfb50f59d2e03cf01c33269c81cb6105ee7446e1/ansible/module_utils/hashivault.py#L111

when there is a python failure behind the scenes, the `Exception` cuts everything except for the last carriage return. example below.

ansible-playbook failure:

`    "msg": "Exception: 'version'",`

if i remove the `try` to something like:

```
def hashiwrapper(function):
    def wrapper(*args, **kwargs):
        result = {"changed": False, "rc": 0}
        result.update(function(*args, **kwargs))
    return wrapper
```

I now get a much more helpful stacktrace:
youll notice it ends in `line 141, in hashivault_secret_engine\nKeyError: 'version'\n` keyword being "version"

```
/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py:18: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses\n  import imp\n/home/vaultadmin/3env/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning,\nTraceback (most recent call last):\n  File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", line 114, in <module>\n    _ansiballz_main()\n  File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/usr/lib/python3.7/imp.py\", line 234, in load_module\n    return load_source(name, filename, file)\n  File \"/usr/lib/python3.7/imp.py\", line 169, in load_source\n    module = _exec(spec, sys.modules[name])\n  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 176, in <module>\n  File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 97, in main\n  File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/ansible_hashivault_secret_engine_payload.zip/ansible/module_utils/hashivault.py\", line 109, in wrapper\n  File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 141, in hashivault_secret_engine\nKeyError: 'version'\n
```

**Comment by drewmullen** [comment]
would love recommendations on a perm fix if you have advice @TerryHowe i can attempt to roll into next PR

**Comment by drewmullen** [comment]
readable output:
```
/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py:18: 
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses\n  
import imp\n
/home/vaultadmin/3env/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. 
  Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  
InsecureRequestWarning,\nTraceback (most recent call last):\n  
File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", 
  line 114, in <module>\n    _ansiballz_main()\n  
File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", 
  line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  
File \"/home/vaultadmin/.ansible/tmp/ansible-tmp-1573831735.4493992-196812719653411/AnsiballZ_hashivault_secret_engine.py\", 
  line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  
  
File \"/usr/lib/python3.7/imp.py\", line 234, in load_module\n    
return load_source(name, filename, file)\n  
File \"/usr/lib/python3.7/imp.py\", line 169, in load_source\n    
module = _exec(spec, sys.modules[name])\n  
File \"<frozen importlib._bootstrap>\", line 630, in _exec\n  
File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n  
File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  
File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 176, in <module>\n  
File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 97, in main\n  
File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/ansible_hashivault_secret_engine_payload.zip/ansible/module_utils/hashivault.py\", 
  line 109, in wrapper\n  
File \"/tmp/ansible_hashivault_secret_engine_payload_6_7udlcd/__main__.py\", line 141, in hashivault_secret_engine\nKeyError: 'version'\n
```

**Comment by drewmullen** [comment]
after spending the day with this... i have 2 possible suggetions

1. eliminate the `hashiwrapper` try block. ansible has native error handling and this attempts to override it.
2. instead of eliminating we can change the generic `Exception` handler and specific error types. this allows ansible to pickup the generic ones which it handles by dumping the stack trace

personally, i dont see much gained in this library by specifically handling the exceptions. but admittedly im not super practiced with the concept.

thoughts?

**Comment by TerryHowe** [comment]
It used to be that leaving those exceptions uncaught caused errors that were very hard to follow, that was the whole point of that.  Perhaps it is no longer needed

**Comment by TerryHowe** [comment]
It seems like such a mixed bag, sometimes I'd see what you are talking about, other times I would get a full stack trace. Anyway, I removed the exception handling for now and we'll give that a try. The stack trace is convenient. Easy to bring back if needed.

https://github.com/TerryHowe/ansible-modules-hashivault/commit/0d6c168f03e5c7b4487c0605f1f91c270f74dd24
